### PR TITLE
SDK-696: Add Document Details Attribute support to Golang SDK

### PR DESCRIPTION
**Pull request by emmas-yoti**

### CODE REVIEW

File: attribute/documentdetailsattribute_test.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	}
+	fmt.Printf(
+		"Document Type: %s, Issuing Country: %s, Document Number: %s, Expiration Date: %s",
+		details.DocumentType,
+		details.IssuingCountry,
+		details.DocumentNumber,
+		details.ExpirationDate,
+	)
+	// Output: Document Type: PASSPORT, Issuing Country: GBR, Document Number: 1234567, Expiration Date: 2022-09-12 00:00:00 +0000 UTC
+}
```
Comment (davidgrayston) [inline attribute/documentdetailsattribute_test.go]: Should we have some more coverage for different strings? e.g. `-` dates and with/without optional attributes.

(Java SDK is a good one to follow for this)

Comment (emmas-yoti) [reply attribute/documentdetailsattribute_test.go]: I think I've addressed this comment now.

File: attribute/documentdetailsattribute.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	}
+
+	dataSlice := strings.Split(data, " ")
+
+	details.DocumentType = dataSlice[0]
+	details.IssuingCountry = dataSlice[1]
+	details.DocumentNumber = dataSlice[2]
+	if len(dataSlice) > 3 {
+		var dateerr error
+		details.expirationDateData, dateerr = time.Parse(documentDetailsDateFormatConst, dataSlice[3])
```
[the remark this answers is not in the document]
Comment (emmas-yoti) [reply attribute/documentdetailsattribute.go]: When the struct is created it's initialized with 'zero values' for all fields. The Zero value for a time.Time object is the 1st January 1AD. Wrapping it with a pointer is the only way to make it 'optional'/give it a sensible nil value.

After the benefit of a good night's rest I have a better solution in my head though.

File: yotiprofile.go [hunk 3]
```diff
@@ -112,3 +113,24 @@ func (p Profile) DocumentImages() (*attribute.ImageSliceAttribute, error) {
 	}
 	return nil, nil
 }
+
+// DocumentDetails returns information about the issuing authority behind
+// a Profile's identity documentation
+func (p Profile) DocumentDetails() (*attribute.DocumentDetailsAttribute, error) {
```
[the remark this answers is not in the document]
Comment (emmas-yoti) [reply yotiprofile.go]: It's not showing as 'Outdated' but the Code view shows this as changed.

File: attribute/documentdetailsattribute.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (echarrod) [reply attribute/documentdetailsattribute.go]: Yep pretty sure it's effectively the same as what your code is doing now! So what you've done is also fine 👍

**Pull request merged**